### PR TITLE
Examples

### PR DESCRIPTION
**Pull request by alejobh**
## Summary

* Added `useLayoutEffect` and `useEffect` example
* Added `useRef` example
* Added `Custom hook: useWindowSize` example
* Added `Context & Persist: Todo List` example
* Added `useRequest` example
* Added `React Query` example

## Screenshots

[Change!] Show the screenshots of the views you modified.

## Trello Card

[Change!] Link to the associated Trello card.

### CODE REVIEW

File: src/app/screens/Dashboard/screens/ReactQueryExample/components/Movies/components/MoviesCache/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { useQueryCache } from 'react-query';
+
+import { GhibliMovieI } from '~utils/types';
+import GhibliMovie from '~components/GhibliMovie';
+
+function MoviesCache({ limit }: { limit: number }) {
+  const cache = useQueryCache();
+
+  const movies: GhibliMovieI[] | undefined = cache.getQueryData(['movies', limit]);
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/ReactQueryExample/components/Movies/components/MoviesCache/index.tsx:10]: Tiene sentido usar el cache directamente? O es para mostrar?

File: src/app/screens/Dashboard/screens/TodoList/components/NewTodo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+
+import FormInput from '~components/FormInput';
+import { useDispatch } from '~contexts/TodosContext';
+import { actionCreators } from '~contexts/TodosContext/reducer';
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/TodoList/components/NewTodo/index.tsx]: I'd move this context to the screens/TodoList folder.

File: src/app/screens/Dashboard/screens/TodoList/components/Todos/components/Todo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { useDispatch } from '~contexts/TodosContext';
+import { actionCreators } from '~contexts/TodosContext/reducer';
+
+import styles from './styles.module.scss';
+
+interface Props {
+  todo: TodoI;
+}
+
+function Todo({ todo }: Props) {
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/TodoList/components/Todos/components/Todo/index.tsx]: I'd name this `TodoItem` and the interface `Todo` o `TodoTask`, what do you think?

File: src/app/screens/Dashboard/screens/UseLayoutEffect/index.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, { useState, useLayoutEffect, useEffect } from 'react';
+import clsx from 'clsx';
+
+import ScreenView from '~components/ScreenView';
+
+import styles from './styles.module.scss';
+
+// eslint-disable-next-line no-magic-numbers
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseLayoutEffect/index.tsx]: I'd disable this rule for the entire project 😅

File: src/app/screens/Dashboard/screens/UseLayoutEffect/index.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+function UseLayoutEffect() {
+  const [value1, setValue1] = useState(0);
+  const [value2, setValue2] = useState(0);
+
+  useLayoutEffect(() => {
+    if (value1 === 0) {
+      setValue1(generateRandom());
+    }
+  }, [value1, value2]);
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseLayoutEffect/index.tsx]: Why is value2 here?

File: src/app/screens/Dashboard/screens/UseRequestExample/index.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  const scrollWhenFetch = useCallback(() => {
+    scrollToBottom();
+  }, []);
+
+  const { data, loading, fetchMore, thereIsMoreData } = useMoviesUseRequest({
+    limit,
+    withPostFetch: scrollWhenFetch
+  });
+
+  // Comment this useEffect, and put limit on dependencies arraay on useMoviesUseRequest (useRequest) to see magic
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseRequestExample/index.tsx]:
```suggestion
  // Comment this useEffect, and put limit on dependencies array on useMoviesUseRequest (useRequest) to see magic
```

**Review by pabloferro — CHANGES_REQUESTED** [review]

File: src/app/screens/Dashboard/screens/ReactQueryExample/components/Movies/components/MoviesCache/index.tsx (hunk 1 above)
Comment (alejobh) [reply src/app/screens/Dashboard/screens/ReactQueryExample/components/Movies/components/MoviesCache/index.tsx:10]: Es para mostrar, y se usa directamente

File: src/app/screens/Dashboard/screens/UseLayoutEffect/index.tsx (hunk 5 above)
Comment (alejobh) [reply src/app/screens/Dashboard/screens/UseLayoutEffect/index.tsx]: :harold:

**Review by pabloferro — APPROVED** [review]

**Pull request merged**